sets_dbs/dp_infos/copy/colab.py

```
@@ -0,0 +1,61 @@
+            urlid = row["urlid"]
+            url = row["url"]
+            # ---
+            if url.find("'") > 0 or url.find('"') > 0:
+                url = escape_string(row["url"])
+            # ---
+            file = row["file"]
+            # ---
+            if file.find("'") > 0 or file.find('"') > 0:
+                file = escape_string(row["file"])
+            # ---
+            insert_statements.append(f"INSERT INTO infos (url, urlid, file) VALUES ('{url}', '{urlid}', '{file}');\n")
```

Ruff (0.14.10)

Possible SQL injection vector through string-based query construction

(S608)